tests/framework/console/controllers/HelpControllerTest.php

```diff
@@ -187,6 +185,37 @@ public function testActionIndexWithServeCommand()
         $this->assertContains('--port, -p: int (defaults to 8080)', $result);
         $this->assertContains('--router, -r: string', $result);
     }
+
+    public function testCommandsFilterOnActionList()
+    {
+        $this->mockApplication([
+            'enableCoreCommands' => false,
+            'controllerNamespace' => 'yiiunit\data\console\controllers',
+        ]);
+        $result = Console::stripAnsiFormat($this->runControllerAction('list'));
+        $this->assertNotContains("fake-empty\n", $result);
+        $this->assertNotContains("fake-no-default\n", $result);
+        $this->assertContains("fake-no-default/index\n", $result);
+        $this->assertContains("fake\n", $result);
```

Comment:
There's no need to test for it. We're checking that action list doesn't contain empty controllers and nothing more.